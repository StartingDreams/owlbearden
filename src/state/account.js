import { fromJS } from 'immutable';

export const firebaseConfig = {
  apiKey: 'AIzaSyDg4Z-8ROBMuRtzU5ThnU1GMd2IWgsttEU',
  authDomain: 'owlbearsden.firebaseapp.com',
  databaseURL: 'https://owlbearsden.firebaseio.com',
  projectId: 'owlbearsden',
  storageBucket: 'owlbearsden.appspot.com',
  messagingSenderId: '149275717879',
};

export const Types = {
  LOGIN: 'owlbearden/LOGIN',
  LOGOUT: 'owlbearden/LOGOUT',
};

export const login = user => ({
  type: Types.LOGIN,
  user,
});

export const logout = () => ({
  type: Types.LOGOUT,
});

export const INITIAL_STATE = fromJS({
  user: null,
  isAuthenticated: false,
});

export default function reducer(state = INITIAL_STATE, action = {}) {
  switch (action.type) {
    case Types.LOGIN:
      return state.withMutations(s => s
        .set('user', fromJS(action.user))
        .set('isAuthenticated', true));
    case Types.LOGOUT:
      return state.withMutations(s => s
        .set('user', fromJS(action.user))
        .set('isAuthenticated', false));
    default:
      return state;
  }
}
