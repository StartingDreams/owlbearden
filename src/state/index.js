import { combineReducers } from 'redux';
import configureStore from './CreateStore';
import ui from './ui';
import casting from '../modules/Casting/ducks/CastingRedux';

export default () => {
  const rootReducer = combineReducers({
    ui,
    casting,
  });

  return configureStore(rootReducer);
};
