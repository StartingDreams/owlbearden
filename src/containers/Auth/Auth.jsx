import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import firebase from 'firebase';
import { withRouter } from 'react-router-dom';
import { login, logout, firebaseConfig } from '../../state/account';

firebase.initializeApp(firebaseConfig);
firebase.auth().setPersistence(firebase.auth.Auth.Persistence.LOCAL);

export class Auth extends React.Component {
  componentDidMount = () => {
    const { dispatch } = this.props;
    firebase.auth().onAuthStateChanged((loginAsUser) => {
      if (loginAsUser) {
        dispatch(login(loginAsUser));
        this.props.history.push('/dashboard');
      } else {
        dispatch(logout());
        this.props.history.push('/login');
      }
    });
  };

  render = () => null;
}

Auth.propTypes = {
  dispatch: PropTypes.func,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

export default withRouter(connect()(Auth));
