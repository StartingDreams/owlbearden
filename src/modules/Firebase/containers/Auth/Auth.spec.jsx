import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import { Auth } from './Auth';
import { login, logout } from '../../../../state/auth';

jest.mock('firebase', () => {
  let callback = null;
  const setPersistence = jest.fn();
  const onAuthStateChanged = (cb) => { callback = cb; };
  const persistence = 'TEST_PERSISTENCE';
  const authReturn = {
    onAuthStateChanged,
    setPersistence,
  };
  const auth = () => (authReturn);

  auth.callback = () => (callback);
  auth.setPersistence = setPersistence;
  auth.persistence = persistence;
  auth.Auth = {
    Persistence: {
      LOCAL: persistence,
    },
  };
  return { auth };
});

const firebase = require('firebase');

describe('<Auth />', () => {
  let dispatch;
  let history;

  beforeEach(() => {
    dispatch = jest.fn();
    history = { push: jest.fn() };
  });

  it('should match the login snapshot', () => {
    const wrapper = shallow(<Auth history={history} dispatch={dispatch} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('persistence should be set', () => {
    const wrapper = shallow(<Auth history={history} dispatch={dispatch} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(firebase.auth().setPersistence).toHaveBeenCalledWith(
      firebase.auth.Auth.Persistence.LOCAL,
    );
  });

  it('login should be dispatched', () => {
    const user = 'USER';
    renderer.create(<Auth history={history} dispatch={dispatch} />);
    const callback = firebase.auth.callback();
    callback(user);
    expect(history.push).toHaveBeenCalledWith('/dashboard');
    expect(dispatch).toHaveBeenCalledWith(login(user));
  });

  it('logout should be dispatched', () => {
    renderer.create(<Auth history={history} dispatch={dispatch} />);
    const callback = firebase.auth.callback();
    callback();
    expect(history.push).toHaveBeenCalledWith('/login');
    expect(dispatch).toHaveBeenCalledWith(logout());
  });
});
