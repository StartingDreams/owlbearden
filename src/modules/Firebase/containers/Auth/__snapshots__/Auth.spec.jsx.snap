// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Auth /> persistence should be set 1`] = `null`;

exports[`<Auth /> should match the login snapshot 1`] = `null`;
